tests/Compiler.test.cpp

```diff
@@ -4475,6 +4480,30 @@ L0: RETURN R0 -1
 )");
 }
 
+TEST_CASE("VectorLiterals")
+{
+    ScopedFastFlag sff("LuauVectorLiterals", true);
+
+    CHECK_EQ("\n" + compileFunction("return Vector3.new(1, 2, 3)", 0, 2, /*enableVectors*/ true), R"(
+LOADK R0 K0 [1, 2, 3]
+RETURN R0 1
+)");
+
+    CHECK_EQ("\n" + compileFunction("print(Vector3.new(1, 2, 3))", 0, 2, /*enableVectors*/ true), R"(
+GETIMPORT R0 1 [print]
+LOADK R1 K2 [1, 2, 3]
+CALL R0 1 0
+RETURN R0 0
+)");
+
+    CHECK_EQ("\n" + compileFunction("print(Vector3.new(1, 2, 3, 4))", 0, 2, /*enableVectors*/ true), R"(
+GETIMPORT R0 1 [print]
+LOADK R1 K2 [1, 2, 3, 4]
+CALL R0 1 0
+RETURN R0 0
+)");
+}
+
```

Comment:
And also maybe a separate test that checks that NaN is preserved correctly.

Reply:
Added unit test for the (0,0,0) vs. (-0,0,0) case.

However, testing NaN is problematic because printf's behavior with NaNs is not standardized. For example, on msvc `return 0/0` generates the following dump:
```
Function 0 (??):
LOADK R0 K0 [-nan(ind)]
RETURN R0 1
```
Clang produces `nan(ind)` and some older msvc versions print `-1.#IND`. So we can't compare the bytecode dump as a string, unless we special case dumpConstant to handle NaNs specifically.

Some alternatives:
* Instead of string compare, execute the bytecode and examine the vector value pushed to the stack by the script. However, it does not feel correct to add a VM dependency to a compiler specific unit test code...
* Compile the bytecode as binary and peek at the correct offset inside the binary. This would add a hard dependency to the bytecode format, which could make the unit test unnecessarily hard to maintain.

Also checking for NaN in binary level is unfortunately not as simple as doing `x != x` because it fails with Fast floating point model. Not sure if Luau is compatible with it though? So we'd probably have to explicitly test the binary value against all possible NaN values.

I also noticed that Compiler.test.cpp does not contain NaN checks for constant folding in general. So perhaps this should be implemented in a more generic way including both numbers and vectors?

All in all, I'm not sure if this is worth the complexity to be included in this PR...

Reply:
Good point re: textual output. We already have NaN tests in vector.lua so I think we're fine on that front.